api/include/opentelemetry/common/stltypes.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <opentelemetry/version.h>
+
+#ifndef HAVE_STDLIB_CPP
+/* Private implementation of standard C++20 types */
+#include "opentelemetry/nostd/stl/span.h"
+#include "opentelemetry/nostd/stl/string_view.h"
+#include "opentelemetry/nostd/stl/variant.h"
+#include "opentelemetry/nostd/stl/utility.h"
+#else
+/* Standard library implementation */
```

Comment:
This seems dangerous. We've floated the idea of libraries vendoring the otel api by keeping a copy of the headers. I'm worried that if we add an ifdef for STL, parts of the build will use it and other parts won't and it will lead to ODR and other problems down the line.

We already have to pay the cost of maintaining nostd/... and I don't know how to reduce that cost but this seems to be increasing it.